### ISSUE TITLE
Can not find config.get_config with dcos 0.4.6

### ISSUE THREAD

**Issue by PaulLiang1** [mesosphere/dcos-spark#33]
Looks like https://github.com/dcos/dcos-cli/commit/2185eb32ee0960059306ce2ff98ff5426924ce8c#diff-0a44a6b7c22f30dd6669ec6caaafa771 have change `util.get_config` to `config.get_config`

`dcos-spark 0.5.14` is producing the following errors as `dcos 0.4.6` had remove that function, downgrading to `dcos 0.4.5` works at the moment. 

```
Traceback (most recent call last):
  File "/Users/Paul/.dcos/subcommands/spark/env/bin/dcos-spark", line 11, in <module>
    sys.exit(main())
  File "/Users/Paul/.dcos/subcommands/spark/env/lib/python2.7/site-packages/dcos_spark/cli.py", line 100, in main
    return show_spark_submit_help()
  File "/Users/Paul/.dcos/subcommands/spark/env/lib/python2.7/site-packages/dcos_spark/cli.py", line 48, in show_spark_submit_help
    return spark_submit.show_help()
  File "/Users/Paul/.dcos/subcommands/spark/env/lib/python2.7/site-packages/dcos_spark/spark_submit.py", line 132, in show_help
    submit_file = spark_file(os.path.join('bin', 'spark-submit'))
  File "/Users/Paul/.dcos/subcommands/spark/env/lib/python2.7/site-packages/dcos_spark/spark_submit.py", line 128, in spark_file
    return os.path.join(spark_dist(), path)
  File "/Users/Paul/.dcos/subcommands/spark/env/lib/python2.7/site-packages/dcos_spark/spark_submit.py", line 71, in spark_dist
    app = spark_app()
  File "/Users/Paul/.dcos/subcommands/spark/env/lib/python2.7/site-packages/dcos_spark/spark_submit.py", line 43, in spark_app
    == service.app_id():
  File "/Users/Paul/.dcos/subcommands/spark/env/lib/python2.7/site-packages/dcos_spark/service.py", line 6, in app_id
    return util.get_config()["spark.app_id"]
AttributeError: 'module' object has no attribute 'get_config'
```

https://github.com/mesosphere/dcos-spark/blob/master/dcos_spark/service.py#L6
should be patched.

**Comment by mgummelt** [comment]
0.4.6 is pretty old, and might be inconsistent with the current `dcos`
package.  Can you upgrade?

On Mon, Jun 27, 2016 at 11:21 PM, Paul Liang <notifications@github.com>
wrote:

> Looks like dcos/dcos-cli@2185eb3#diff-0a44a6b7c22f30dd6669ec6caaafa771
> <https://github.com/dcos/dcos-cli/commit/2185eb32ee0960059306ce2ff98ff5426924ce8c#diff-0a44a6b7c22f30dd6669ec6caaafa771>
> have change util.get_config to config.get_config
>
> dcos-spark 0.4.6 is producing the following errors as it can not be
> found, downgrading to 0.4.5 works at the moment.
>
> Traceback (most recent call last):
>   File "/Users/Paul/.dcos/subcommands/spark/env/bin/dcos-spark", line 11, in <module>
>     sys.exit(main())
>   File "/Users/Paul/.dcos/subcommands/spark/env/lib/python2.7/site-packages/dcos_spark/cli.py", line 100, in main
>     return show_spark_submit_help()
>   File "/Users/Paul/.dcos/subcommands/spark/env/lib/python2.7/site-packages/dcos_spark/cli.py", line 48, in show_spark_submit_help
>     return spark_submit.show_help()
>   File "/Users/Paul/.dcos/subcommands/spark/env/lib/python2.7/site-packages/dcos_spark/spark_submit.py", line 132, in show_help
>     submit_file = spark_file(os.path.join('bin', 'spark-submit'))
>   File "/Users/Paul/.dcos/subcommands/spark/env/lib/python2.7/site-packages/dcos_spark/spark_submit.py", line 128, in spark_file
>     return os.path.join(spark_dist(), path)
>   File "/Users/Paul/.dcos/subcommands/spark/env/lib/python2.7/site-packages/dcos_spark/spark_submit.py", line 71, in spark_dist
>     app = spark_app()
>   File "/Users/Paul/.dcos/subcommands/spark/env/lib/python2.7/site-packages/dcos_spark/spark_submit.py", line 43, in spark_app
>     == service.app_id():
>   File "/Users/Paul/.dcos/subcommands/spark/env/lib/python2.7/site-packages/dcos_spark/service.py", line 6, in app_id
>     return util.get_config()["spark.app_id"]
> AttributeError: 'module' object has no attribute 'get_config'
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/dcos-spark/issues/33>, or mute the thread
> <https://github.com/notifications/unsubscribe/AAJYhoCgb5n6oOEazt3j2hKako43mZIsks5qQL1dgaJpZM4I_xwG>
> .
>



-- 
Michael Gummelt
Software Engineer
Mesosphere

**Comment by PaulLiang1** [comment]
@mgummelt 

Sorry for the confusion, the 0.4.6 was for `dcos-cli` available here:
https://github.com/dcos/dcos-cli/releases/tag/0.4.6

The latest release available on `pip` is `0.4.6` where `util.get_config` had been removed. 
https://github.com/dcos/dcos-cli/commit/2185eb32ee0960059306ce2ff98ff5426924ce8c#diff-0a44a6b7c22f30dd6669ec6caaafa771L191

and placed into `config.get_config`
https://github.com/dcos/dcos-cli/blob/master/dcos/config.py#L39-L57

`dcos-spark` is still referencing `get_config` from `util` which will break on the latest release. 

This error could be reproduced by:
- Install a fresh DCOS CLI using `install.sh` following instructions in `https://docs.mesosphere.com/1.7/usage/cli/install/`
- Install DCOS Spark CLI  using `dcos packages install spark --cli`,  To ensure a fresh install, remove `$HOME/.dcos`

**Comment by mgummelt** [comment]
Yea, it's a bug.  I'll try to get a new spark release out soon.  In the
mean time, you could try to monkey patch your dcos-spark.  Looks like you
already found the source.

On Tue, Jun 28, 2016 at 8:39 PM, Paul Liang <notifications@github.com>
wrote:

> @mgummelt <https://github.com/mgummelt>
>
> Sorry for the confusion, the 0.4.6 was for dcos-cli available here:
> https://github.com/dcos/dcos-cli/releases/tag/0.4.6
>
> The latest release available on pip is 0.4.6 where util.get_config had
> been removed.
> dcos/dcos-cli@2185eb3#diff-0a44a6b7c22f30dd6669ec6caaafa771L191
> <https://github.com/dcos/dcos-cli/commit/2185eb32ee0960059306ce2ff98ff5426924ce8c#diff-0a44a6b7c22f30dd6669ec6caaafa771L191>
>
> and placed into config.get_config
> https://github.com/dcos/dcos-cli/blob/master/dcos/config.py#L39-L57
>
> dcos-spark is still referencing get_config from util which will break on
> the latest release.
>
> This error could be reproduced by:
>
>    - Install a fresh DCOS CLI using install.sh following instructions in
>    https://docs.mesosphere.com/1.7/usage/cli/install/
>    - Install DCOS Spark CLI using dcos packages install spark --cli, To
>    ensure a fresh install, remove $HOME/.dcos
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/dcos-spark/issues/33#issuecomment-229248499>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAJYhj6M6NY0pT_Rwsg6KKKTIOQmPj_uks5qQekHgaJpZM4I_xwG>
> .
>



-- 
Michael Gummelt
Software Engineer
Mesosphere

**Comment by PaulLiang1** [comment]
Yeah usually I would made a pull request with fix of freezing the version number in the dependencies, however `dcos-cli` and `dcos-spark` seem to be using a different wa

**Comment by PaulLiang1** [comment]
Accidentally clicked on the touch pad..

**Comment by mgummelt** [comment]
A fix is pushed: https://github.com/mesosphere/universe/pull/532

Please allow 10-20 minutes for propagation to universe

On Wed, Jun 29, 2016 at 10:41 AM, Michael Gummelt <mgummelt@mesosphere.io>
wrote:

> Yea, it's a bug.  I'll try to get a new spark release out soon.  In the
> mean time, you could try to monkey patch your dcos-spark.  Looks like you
> already found the source.
>
> On Tue, Jun 28, 2016 at 8:39 PM, Paul Liang <notifications@github.com>
> wrote:
>
>> @mgummelt <https://github.com/mgummelt>
>>
>> Sorry for the confusion, the 0.4.6 was for dcos-cli available here:
>> https://github.com/dcos/dcos-cli/releases/tag/0.4.6
>>
>> The latest release available on pip is 0.4.6 where util.get_config had
>> been removed.
>> dcos/dcos-cli@2185eb3#diff-0a44a6b7c22f30dd6669ec6caaafa771L191
>> <https://github.com/dcos/dcos-cli/commit/2185eb32ee0960059306ce2ff98ff5426924ce8c#diff-0a44a6b7c22f30dd6669ec6caaafa771L191>
>>
>> and placed into config.get_config
>> https://github.com/dcos/dcos-cli/blob/master/dcos/config.py#L39-L57
>>
>> dcos-spark is still referencing get_config from util which will break on
>> the latest release.
>>
>> This error could be reproduced by:
>>
>>    - Install a fresh DCOS CLI using install.sh following instructions in
>>    https://docs.mesosphere.com/1.7/usage/cli/install/
>>    - Install DCOS Spark CLI using dcos packages install spark --cli, To
>>    ensure a fresh install, remove $HOME/.dcos
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/mesosphere/dcos-spark/issues/33#issuecomment-229248499>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/AAJYhj6M6NY0pT_Rwsg6KKKTIOQmPj_uks5qQekHgaJpZM4I_xwG>
>> .
>>
>
>
>
> --
> Michael Gummelt
> Software Engineer
> Mesosphere
>



-- 
Michael Gummelt
Software Engineer
Mesosphere

**Comment by PaulLiang1** [comment]
Thanks!
